app/javascript/packages/document-capture/components/in-person-location-post-office-search-step.tsx

```diff
@@ -77,15 +86,82 @@ const requestUspsLocations = async (address: LocationQuery): Promise<FormattedLo
   return formatLocation(response);
 };
 
-function InPersonLocationPostOfficeSearchStep({ onChange, toPreviousStep, registerField }) {
+const ADDRESS_SEARCH_URL = '/api/addresses';
+
+function requestAddressCandidates(unvalidatedAddressInput: string): Promise<Location[]> {
+  return request<Location[]>(ADDRESS_SEARCH_URL, {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    json: { address: unvalidatedAddressInput },
+  });
+}
+
+function useUspsLocations() {
+  // raw text input that is set when user clicks search
+  const [addressQuery, setAddressQuery] = useState('');
+  const [fieldValidation, setFieldValidation] = useState<RefObject<HTMLFormElement>>();
   const { t } = useI18n();
+  const handleAddressSearch = useCallback((event, unvalidatedAddressInput, validatedFieldRef) => {
+    event.preventDefault();
+    validatedFieldRef.current?.setCustomValidity('');
+    validatedFieldRef.current?.reportValidity();
+    setFieldValidation(validatedFieldRef);
```

Comment:
**DO NOT** keep a direct refence to the child form's HTML element. This is a use case for a proxy object created via `useImperativeHandle`. Also **DO NOT** pass the HTML element from that function either, `setCustomValidity` and `reportValidity` should ultimately be called from the `AddressSearch` component (inside of the proxy object).

Reply:
This is b/c React depends very heavily on one-way data flow and relative decoupling of parent vs. child components. Here we're essentially reaching down to the child component's own child component in order to update the form. 

What's important here is **that is the child component's concern**. If the input gets replaced or updated by the direct child, then this ancestor is likely to either 1) fail in performing its necessary function, or 2) could update the form in an undesirable manner. Most likely, the reference would become stale and we could be attempting to update an element that no longer exists in the form.

React is abstracting many parts of how the DOM gets updated, and it's best that direct DOM function calls like `setCustomValidity` and `reportValidity` remain as close to the component rendering the actual input as possible. That way we can reduce the blast radius when there's a difference in the DOM API or if we need to provide additional validation-related functionality.

Furthermore, this coupling reduces the portability of the address component because now you need to implement a specific type of function handle to get the reference instead of a standard React API like refs.